wave_lang/kernel/wave/water.py

```diff
@@ -446,3 +446,271 @@ def add_opt(pipeline):
 
     with module.context:
         return Module.parse(result)
+
+
+def _find_water_python_path():
+    """Find the Water Python bindings path dynamically."""
+    import sys
+    from pathlib import Path
+
+    debug_info = []  # Collect debug info for troubleshooting
+
+    # Try to find Water build directory using the same logic as get_water_opt
+    try:
+        water_opt_path = Path(get_water_opt())
+        debug_info.append(f"Found water-opt at: {water_opt_path}")
+
+        # water-opt is typically in build/bin/water-opt, so python packages are in build/python_packages
```

Comment:
Nit: it may be typical for you using it from the build tree, but it is not the general use case, which would be wheels installation